cluster-autoscaler/provisioningrequest/provreqclient/client.go

```diff
@@ -123,6 +125,21 @@ func (c *ProvisioningRequestClient) FetchPodTemplates(pr *v1beta1.ProvisioningRe
 	return podTemplates, nil
 }
 
+// UpdateProvisioningRequest updates the given ProvisioningRequest CR by propagating the changes using the ProvisioningRequestInterface and returns the updated instance or the original one in case of an error.
+func (c *ProvisioningRequestClient) UpdateProvisioningRequest(pr *v1beta1.ProvisioningRequest) (*v1beta1.ProvisioningRequest, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), provisioningRequestClientCallTimeout)
+	defer cancel()
+
+	prCopy := pr.DeepCopy()
+	prCopy.Spec = v1beta1.ProvisioningRequestSpec{}
+	updatedPr, err := c.client.AutoscalingV1beta1().ProvisioningRequests(prCopy.Namespace).UpdateStatus(ctx, prCopy, metav1.UpdateOptions{})
+	if err != nil {
+		return pr, err
+	}
+	klog.V(4).Infof("Updated ProvisioningRequest %s/%s,  status: %q,", updatedPr.Namespace, updatedPr.Name, updatedPr.Status)
```

Comment:
If this happens every loop (which I suspect it does), that should not be V(4). Logging per-workload per-loop is an anti-pattern which makes CA logs impossible to read in clusters with thousands of nodes & workloads. 
If a log is really necessary we prefer to use klogx to limit maximum number of logs per loop, but here it would be slightly more complicated as you'd need to keep the counter between calls and reset it at the start of the loop - so I'd rather just reduce to V(5).

Reply:
>If this happens every loop ...

It shouldn't, the conditions might be updated once in 10 mins for 1 ProvReq  and as of now we process 1 ProvReq for 1 loop.

Reply:
IIUC:
* Accepted, Failed, and BookingExpired updates will only happen once per ProvReq.
* Provisioned updates will happen at most once per 10 mins per ProvReq.

@yaroslava-serdiuk is my understanding correct? If so, I think the logging level should be fine for now, although it'll be easy to miss if we start using this method more frequently in the future.

Reply:
Ack, this is ok IMO